### PR TITLE
Expose stream parameter in public reduction APIs

### PR DESCRIPTION
**Pull request by srinivasyadav18**
## Description

Add stream parameter to public reduction APIs:

- `reduce()`
- `segmented_reduce()`
- `scan()`
- `minmax()`

Reference #13744 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by davidwendt — APPROVED** [review]

**Review by mhaseeb123 — APPROVED** [review]
LGTM!

**Comment by srinivasyadav18** [comment]
Hello,
3 tests are still failing. I am looking into it ;-). Any leads would be appreciated. Thanks!

File: cpp/tests/streams/reduction_test.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  cudf::reduce(input,
+               *cudf::make_sum_aggregation<cudf::reduce_aggregation>(),
+               cudf::data_type(cudf::type_id::INT32),
+               cudf::test::get_default_stream());
+}
+
+TEST_F(ReductionTest, ReductionSumInit)
+{
+  cudf::test::fixed_width_column_wrapper<int> input({1, 2, 3, 4, 5, 6, 7, 8, 9, 10});
+  auto const init_scalar = cudf::make_fixed_width_scalar<int>(3);
```
Comment (davidwendt) [inline cpp/tests/streams/reduction_test.cpp:41]:
```suggestion
  auto const init_scalar = cudf::make_fixed_width_scalar<int>(3, cudf::test::get_default_stream());
```
This is likely why the tests are failing.

File: cpp/tests/streams/reduction_test.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+TEST_F(ReductionTest, SegmentedReductionSumScalarInit)
+{
+  auto const input     = cudf::test::fixed_width_column_wrapper<int>{{1, 2, 3, 1, 0, 3, 1, 0, 0, 0},
+                                                                     {1, 1, 1, 1, 0, 1, 1, 0, 0, 0}};
+  auto const offsets   = std::vector<cudf::size_type>{0, 3, 6, 7, 8, 10, 10};
+  auto const d_offsets = cudf::detail::make_device_uvector_async(
+    offsets, cudf::test::get_default_stream(), rmm::mr::get_current_device_resource());
+  auto const init_scalar = cudf::make_fixed_width_scalar<int>(3);
```
Comment (davidwendt) [inline cpp/tests/streams/reduction_test.cpp:73]:
```suggestion
  auto const init_scalar = cudf::make_fixed_width_scalar<int>(3, cudf::test::get_default_stream());
```

**Review by davidwendt — CHANGES_REQUESTED** [review]

**Comment by srinivasyadav18** [comment]
Thank you @davidwendt. Your suggestions worked out!

**Review by davidwendt — APPROVED** [review]

**Review by PointKernel — APPROVED** [review]
LGTM

@srinivasyadav18 Welcome and thanks for the contribution!

Just FYI, CI is broken right now thus it will take a bit longer to merge the current PR.

**Pull request merged**